webpack/assets/javascripts/react_app/redux/actions/common/forms.js

```diff
@@ -47,13 +47,24 @@ const verifyProps = (item, values) => {
   }
 };
 
-export const submitForm = ({ item, url, values, message, method = 'post' }) => {
-  verifyProps(item, values);
+export const submitForm = ({
+  item,
+  url,
+  values: params,
+  message,
+  method = 'post',
+  headers,
+  apiActionTypes: actionTypes,
+}) => {
+  verifyProps(item, params);
   return async dispatch => {
-    try {
-      const { data } = await API[method](url, values);
+    const uniqueAPIKey = `${item.toUpperCase()}_FORM_SUBMITTED`;
+
+    const handleError = error => onError(error);
+
+    const handleSuccess = ({ data }) => {
       dispatch({
-        type: `${item.toUpperCase()}_FORM_SUBMITTED`,
+        type: uniqueAPIKey,
```

Comment:
@xprazak2 should we still dispatch this action after the API call secceed?
or maybe we can listen to the API middleware triggered actions, e.g:
`item_FORM_SUBMITTED_REQUEST`
`item_FORM_SUBMITTED_SUCCESS`
`item_FORM_SUBMITTED_FAILURE`

I wonder if we should do it in another PR? as I don't know for sure where is it used except for `bookmarks`